File: 2020/2020W35/PreppindataW35.py
import pandas as pd

xlsx = pd.ExcelFile("F:/Data/Input Week 35.xlsx")
sales = pd.read_excel(xlsx,'No RowID')   

sales['Store'] = sales['Store'].ffill()
sales = sales.dropna(subset=['Store',sales.columns[1]])
sales['RowID'] = sales.index
sales['MaxRowID'] = sales['RowID'].groupby(sales['Store']).transform('max')
sales['Type'] = sales.apply(lambda x: 'Difference' if x['RowID'] == x['MaxRowID'] 
                            else 'Target' if x['RowID'] == x['MaxRowID']-1 else 'Sales', axis=1)
sales = sales.drop(['RowID','MaxRowID'], axis=1)
sales = sales.melt(id_vars=['Store','Type'], value_name='Value', var_name='Month')
sales['Month'] = sales['Month'].apply(lambda x: pd.to_datetime(x[0:3]+' 01 2020', format='%b %d %Y').date()) 

sales = sales.pivot_table(index=['Store','Month'],columns='Type', values='Value', aggfunc=max).reset_index()
sales = sales[['Store', 'Month', 'Sales', 'Target', 'Difference']]

File: 2020/2020W35/preppindataw35.R
library(dplyr)
library(readxl)
library(tidyr)
library(zoo)

Sys.setlocale("LC_ALL","English")

final <- read_excel("F:/Data/Input Week 35.xlsx", sheet = 'No RowID') %>%
  drop_na(!!names(.[2])) %>%
  do(na.locf(.)) %>%
  group_by(`Store`) %>%
  mutate('Type' = case_when(row_number()==1 ~ 'Sales',
                            row_number()==2 ~ 'Target',
                            TRUE ~ 'Difference')) %>%
  pivot_longer(., cols=matches('Sales$'), names_to='Month', values_to='Value') %>%
  mutate('Month' = as.Date(paste(substring(`Month`,1,3),'01','2020'), format='%b %d %Y')) %>%
  pivot_wider(., names_from='Type', values_from='Value')

View(final)